src/gradualizer_lib.erl

```diff
@@ -120,6 +125,67 @@ pick_value(?type(range, [{_TagLo, _, neg_inf}, {_TagHi, _, Hi}])) ->
 pick_value(?type(range, [{_TagLo, _, Lo}, {_TagHi, _, _Hi}])) ->
     Lo.
 
+%% Prettyprints a ?type(record)
+%%
+%% What to expect:
+%% ```
+%%   #record_name{
+%%     field_one = #nested_record{
+%%       nested_field = 1
+%%     },
+%%     field_two = some_atom
+%%   }
+%% ```
+prettyprint(Record) ->
+    Doc =
+        case pp_record(Record) of
+            X when is_list(X) -> lists:foldr(fun prettypr:above/2, prettypr:empty(), X);
+            X -> X
+        end,
+    prettypr:format(Doc).
+
+-spec pp_record(gradualizer_type:abstract_type()) -> prettyprint:document() | [prettyprint:document()].
+pp_record(?type(record, [{atom, _, RecName} | Fields])) ->
+    PPFields = pp_record_fields(Fields),
```

Comment:
I don't quite see why you would have to implement your own pretty-printing for records. It's already implemented in `erl_pp` (abstract form) and in `io_lib:format("~p", [Rec])`

Reply:
Thanks, I didn't know about `erl_pp`.